### PR TITLE
Upgrade operator to 0.99.0

### LINKED ISSUE #1167: Collector CRD 0.99.0 requires templated conversion webhook
Created from https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1164 

The conversion webhook defined in the collector CRD should be templated because it references the operator webhook service which can be dynamically set:

```bash
helm install --namespace=my-opentelemetry-operator-system --create-namespace my-opentelemetry-operator ./charts/opentelemetry-operator
```

```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    cert-manager.io/inject-ca-from: opentelemetry-operator-system/opentelemetry-operator-serving-cert
    controller-gen.kubebuilder.io/version: v0.14.0
  name: opentelemetrycollectors.opentelemetry.io
spec:
  conversion:
    strategy: Webhook
    webhook:
      clientConfig:
        service:
          name: {missing my-}opentelemetry-operator-webhook
          namespace: {missing my}opentelemetry-operator-system
          path: /convert
      conversionReviewVersions:
      - v1alpha1
      - v1beta1
```

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Resolves #1167, alternative to #1164. 

In order to support using the same certificates for the conversion webhook as we do for other webhooks, I needed a way to inject the CA bundle into the CRD templates. Unfortunately, in Helm all values are scoped to a file, and it's impossible to cache any of them, other than via resources in the cluster. What I ended up doing is moving the CRD templates to a `conf/crds` directory, and including them as templates via the `tpl` function. This means that for the time being, they're declared in the webhook file. I'd like to change this in a follow up, probably by moving the webhooks to `conf/webhooks` and then including them the same way I did for CRDs, but that's up for discussion.

### CODE REVIEW

**Review by pavolloffay — APPROVED** [review]

**Comment by swiatekm-sumo** [comment]
Please don't merge this before #1175. We've decided to do a separate release with just the CRD transition, and then the upgrade to operator 0.99.0.

**Comment by TylerHelmuth** [comment]
> I'd like to change this in a follow up, probably by moving the webhooks to conf/webhooks and then including them the same way I did for CRDs, but that's up for discussion.

Agreed, created https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1181

**Review by TylerHelmuth — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]
What's the best way to ensure that users see the documentation in the Upgrading.md? I am just assuming there will be users that won't know to look there and would love to call out those instructions really clearly.

**Comment by jvoravong** [comment]
> What's the best way to ensure that users see the documentation in the Upgrading.md? I am just assuming there will be users that won't know to look there and would love to call out those instructions really clearly.

I might suggest adding a smaller note in CHANGLOG.md that reference the full note in UPGRADING.md.

**Comment by TylerHelmuth** [comment]
We don't have changelog files since each commit is a release - I will manually update the release notes with instructions.

**Review by JaredTan95 — APPROVED** [review]

**Review by Allex1 — APPROVED** [review]

**Review by TylerHelmuth — CHANGES_REQUESTED** [review]
Need https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1183 first

**Review by TylerHelmuth — COMMENTED** [review]
Once we have the docs update we'll be good to go

File: charts/opentelemetry-operator/README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # OpenTelemetry Operator Helm Chart
 
+> [!WARNING]  
+> Version 0.58.0 of this Chart includes a new version of the `OpenTelemetryCollector` CRD. See [this document][v1beta1_migration] for upgrade instructions. Please make sure you also follow the [upgrade instructions](./UPGRADING.md#0560-to-0570) for 0.57.0.
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/README.md:4]:
```suggestion
> Version 0.58.0 of this Chart includes a new version of the `OpenTelemetryCollector` CRD. See [this document][v1beta1_migration] for upgrade instructions for the new Operator CRD. Please make sure you also follow the [helm upgrade instructions](./UPGRADING.md#0560-to-0570) for helm chart 0.57.0.
```

File: charts/opentelemetry-operator/UPGRADING.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Upgrade guidelines
 
+## 0.57.0 to 0.58.0
+
+OpenTelemetry Operator [0.99.0](https://github.com/open-telemetry/opentelemetry-operator/releases/tag/v0.99.0) includes a new version of the `OpenTelemetryCollector` CRD. See [this document][v1beta1_migration] for upgrade instructions. Please make sure you also follow the [upgrade instructions](./UPGRADING.md#0560-to-0570) for 0.57.0.
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/UPGRADING.md:5]:
```suggestion
OpenTelemetry Operator [0.99.0](https://github.com/open-telemetry/opentelemetry-operator/releases/tag/v0.99.0) includes a new version of the `OpenTelemetryCollector` CRD. See [this document][v1beta1_migration] for upgrade instructions for the new Operator CRD. Please make sure you also follow the [helm upgrade instructions](./UPGRADING.md#0560-to-0570) for helm chart 0.57.0.
```

**Pull request merged**